### ISSUE TITLE
Upgrade to Swift2

### ISSUE THREAD

**Issue by ghost** [huyouare/SwiftParseChat#10]

**Comment by MHX792** [comment]
+1 Yes, definitley need this!

**Comment by pimpaocom** [comment]
+1!

**Comment by LualdiD** [comment]
+1

**Comment by lmick002** [comment]
+1

**Comment by lmick002** [comment]
+1

**Comment by moh1122** [comment]
?

**Comment by ericpeters0n** [comment]
+1

**Comment by moh1122** [comment]
not working

**Comment by zzglfms** [comment]
Anybody working on that?

**Comment by webbersky** [comment]
https://github.com/webbersky/SwiftParseChat-master
You're welcome (:

**Comment by kevinlee85** [comment]
+1

**Comment by korrio** [comment]
@webbersky You save my life !!

**Comment by lmick002** [comment]
Hey after finding out that client pushes can be relatively not secure , i implemented a parse cloud code function to send the pushes. let me know if anyone is interested in it. thanks

**Comment by MHX792** [comment]
I'd be interested! What do you mean by "not secure"? Is it unreliable in terms of sending push notifications or in terms of security?

**Comment by lmick002** [comment]
yeah in terms of security http://blog.parse.com/learn/engineering/the-dangerous-world-of-client-push/

**Comment by MHX792** [comment]
Thanks for the link. Would you be as kind as to share your code with us? :+1:

**Comment by MHX792** [comment]
Still intending to share it? Would like to see how to do it with cloud code.

**Comment by lmick002** [comment]
sorry ive been busy i will share =)

**Comment by MHX792** [comment]
Any news on when you are planning to share it?

**Comment by MHX792** [comment]
@lmick002 Still planning to share it? :)

**Comment by lmick002** [comment]
@MHX792 sure give me one min :)) :+1:

**Comment by lmick002** [comment]
@MHX792 here it is https://github.com/lmick002/ParseChat-Push

**Comment by MHX792** [comment]
Thanks a lot! Going to try it soon!


Mit freundlichen Grüßen
Alper Beser

**Comment by MHX792** [comment]
Thanks a lot! Going to try it soon!

**Comment by jonathanwuki** [comment]
I ported this application to Swift 2, however it does lose Facebook login functionality. You can view the repo [here](https://github.com/insleep/SwiftParseChat).
